src/main/webapp/app/tasks/local-task.service.ts

```diff
@@ -32,9 +32,24 @@ export class LocalTaskService implements TaskService {
     return of(null);
   }
 
-  private readTasks(): Task[] {
+  private readTasks(query?:string): Task[] {
+
     const tasks = localStorage.getItem(LocalTaskService.STORAGE_KEY);
-    return tasks ? JSON.parse(tasks) : [];
+    const tasksList = JSON.parse(tasks);
+    const searchResult = [];
+
+    if (tasks && !query) {
+      return tasksList;
+    } else if (query) {
+      tasksList.forEach((task)=> {
+        if (task.name.toLowerCase().includes(query.toLowerCase())) {
+          searchResult.push(task);
+        }
+      });
+      return searchResult;
+    } else {
+      return [];
```

Comment:
you don't need the else branch if you simply call `return searchResult;` at the end of the function, that way you can also save yourself the trouble to do the return in line 49 as well